### DISCUSSION TITLE
How to allow anyone to access a table?

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#7039] [Questions]
I am trying to allow anyone to access a table without signing in (it works when signed in). How do I do this? I made a policy "enable read access to everyone" but it doesn't work.

**Comment by GaryAustin1** [comment]
Is your actual policy "true"?
```
FOR SELECT USING (
  true
);
```
Then do you have your anon key in the SB client initialization?
Does it work with RLS off?

**Reply by ghost** [reply]
Yeah, it's true. I don't know what SB client initialization is, but I pasted a key somewhere. It does not work with RLS off.

**Reply by GaryAustin1** [reply]
Can you show your query to table?  Can you see the network response for any error?
 Not knowing the language you are using, this is js init I was mentioning:
 const supabase = createClient('https://xyzcompany.supabase.co', 'public-anon-key')

**Reply by ghost** [reply]
I'm really sorry, but I have no clue what any of that means. I'm very new.

I am using a [Supabase integration for the Godot game engine](https://github.com/fenix-hub/godot-engine.supabase).

The error is "empty >> JWSError (CompactDecodeError Invalid number of parts: Expected 3 parts; got 1): (empty)"
My query is `SupabaseQuery.new().from("creators").select(["*"])` which converts to "creators?select=*".

I have a table called "creators" in Supabase.

**Reply by GaryAustin1** [reply]
Well your error could be caused by not having anon key set at init time.   When you don't have a user signed, the anon jwt key gets passed in the database call.  
I notice this issue at that site https://github.com/fenix-hub/godot-engine.supabase/issues/11 which would cause what you are seeing, but that claims it was fixed.

**Reply by ghost** [reply]
Huh. I'm pretty sure it is set at init time. I'll look into that issue.

**Reply by GaryAustin1** [reply]
This image is request headers with network call and signed in user (note the longer jwt in authorization header):
![image](https://user-images.githubusercontent.com/54564956/170886593-10ff1f69-8b8c-454e-9369-aeb0695db445.png)
This image is request headers with network call and no signed in user:
![image](https://user-images.githubusercontent.com/54564956/170886620-4d915126-117f-41ce-8f68-544a2d049e60.png)
Note how the two headers have the same token in the 2nd one.  Supabase-js is being used and it knows to copy the anon jwt key to the authorization header when there is not signed in user.
I'm not sure what that repository is doing for that, but you can look at the request and if both headers are not set the same for anon case, that is the issue.

**Reply by GaryAustin1** [reply]
I glanced at the code in the repository.  Are you calling: sign_in_anonymous()?
https://github.com/fenix-hub/godot-engine.supabase/blob/04c3759ff6e8718d81fc86d3abdec9bc14f195e8/addons/supabase/Auth/auth.gd#L148
It looks like the code may not set the token as default and that call sets it.

**Reply by ghost** [reply]
Thank you for the suggestion. I called the function, but got the error "invalid_request >> refresh token required".

**Reply by GaryAustin1** [reply]
At this point you really need to work with people using the repository you are using (it has not been updated in 7 months).  I have no idea if it even works.

It makes no sense for a refresh token to be required for an anonymous request.  So either you are doing something wrong with their code, or their code has a major issue with anon user.

For SB to work you need network requests to have the apikey/authorization headers looking similar to my images above.  You don't comment if they are. But if not then SB will not work correctly for anon user access.

**Reply by ghost** [reply]
Okay, I'll contact the creator of the repo!
